doc/man/8/ceph-objectstore-tool.rst

```diff
@@ -0,0 +1,492 @@
+
+ Syntax::
+
+   ceph-objectstore-tool --data-path $PATH_TO_OSD --pgid $PG_ID $OBJECT  set-attrs $KEY < $OBJECT_ATTRS_FILE_NAME
+
+ Example::
+
+   [root@osd ~]# ceph-objectstore-tool --data-path /var/lib/ceph/osd/ceph-0 --pgid 0.1c '{"oid":"zone_info.default","key":"","snapid":-2,"hash":235010478,"max":0,"pool":11,"namespace":""}' set-attrs "oid" < zone_info.default.attr.txt
+
+ Remove an object’s attributes:
+
+ Syntax::
+
+   ceph-objectstore-tool --data-path $PATH_TO_OSD --pgid $PG_ID $OBJECT rm-attrs $KEY
+
+ Example::
+
+   [root@osd ~]# ceph-objectstore-tool --data-path /var/lib/ceph/osd/ceph-0 --pgid 0.1c '{"oid":"zone_info.default","key":"","snapid":-2,"hash":235010478,"max":0,"pool":11,"namespace":""}' rm-attrs "oid"
+
+
+Options
+=======
+
+.. option:: --help          
+
+   produce help message
+
+.. option:: --type arg        
+
+   Arg is one of [bluestore (default), filestore, memstore]. This option is needed only if the tool can't tell the type from --data-path.
+ 
+.. option:: --data-path arg
+
+   path to object store, mandatory
+   
+.. option:: --journal-path arg
+
+   path to journal, use if tool can't find it
+   
+.. option:: --pgid arg
+
+   PG id, mandatory for info, log, remove, export, export-remove, mark-complete, trim-pg-log, and mandatory for apply-layout-settings if --pool is not specified
+                             
+.. option:: --pool arg
+
+   Pool name, mandatory for apply-layout-settings if --pgid is not specified
+
+.. option:: --op arg
+
+   Arg is one of [info, log, remove, mkfs, fsck, repair, fuse, dup, export, export-remove, import, list, fix-lost, list-pgs, dump-journal, dump-super, meta-list, get-osdmap, set-osdmap, get-inc-osdmap, set-inc-osdmap, mark-complete, reset-last-complete, apply-layout-settings, update-mon-db, dump-export, trim-pg-log]
+
+.. option:: --epoch arg
+
+   epoch# for get-osdmap and get-inc-osdmap, the current epoch in use if not specified
+
+.. option:: --file arg             
+   
+   path of file to export, export-remove, import, get-osdmap, set-osdmap, get-inc-osdmap or set-inc-osdmap
+
+.. option:: --mon-store-path arg
+
+   path of monstore to update-mon-db
+
+.. option:: --fsid arg
+
+   fsid for new store created by mkfs
+
+.. option:: --target-data-path arg
+
+   path of target object store (for --op dup)
+   
+.. option:: --mountpoint arg
+
+   fuse mountpoint
+
+.. option:: --format arg (=json-pretty) 
+
+   Output format which may be json, json-pretty, xml, xml-pretty
+
+.. option:: --debug
+
+   Enable diagnostic output to stderr
+
+.. option:: --force
+
+   Ignore some types of errors and proceed with operation - USE WITH CAUTION: CORRUPTION POSSIBLE NOW OR IN THE FUTURE
+
+.. option:: --skip-journal-replay
+
+   Disable journal replay
+
+.. option:: --skip-mount-omap
+
+   Disable mounting of omap
+
+.. option:: --head
+
+   Find head/snapdir when searching for objects by name
+
+.. option:: --dry-run
+
+   Don't modify the objectstore
```

Comment:
In reality unless you also specify --skip-journal-replay the objectstore is written to.  I don't know exactly how to put, but technically "Don't modify the objectstore" is misleading.  It is a dry run of the operation requested.